level2.src.html

```diff
@@ -84,29 +84,90 @@ <h2 id="algorithms">Algorithms</h2>
     <h3 id="upgrade-algorithm">Upgrade |request| to an <i lang="la">a priori</i>
     authenticated URL as mixed content, if appropriate</h3>
 
-    TODO. This algorithm is called from Fetch before blocking requests as mixed content. It checks if the request is optionally-blockable mixed content, and if so rewrites the scheme to HTTPS.
+    Note: The Fetch specification will hook into this algorithm to upgrade optionally-blockable
+    mixed content to HTTPS.
+
+    Given a <a>Request</a> <var>request</var>, this algorithm will rewrite
+    its <a for="request">url</a> if the request is deemed to be optionally-blockable mixed content,
+    via the following algorithm:
+
+    <ol>
+      <li>
+        If one or more of the following conditions is met, return without modifying <var>request</var>:
+        <ol>
+          <li>
+            [[mixed-content#categorize-settings-object]] returns "<code>Does Not Restrict Mixed Security
+            Contents</code>" when applied to <var>request</var>'s <a for="request">client</a>.
+          </li>
+          <li>
+            <var>request</var>'s <a for="request">mode</a> is <code>CORS</code>
```

Comment:
Why would we skip upgrading CORS requests? It seems like `fetch("http://insecure.data/", { mode: "cors" })` is just as reasonable to poke at as `fetch("http://insecure.data/", { mode: "no-cors" })`

Reply:
Skipping upgrading because they're blocked as active mixed content (Step 3 of https://w3c.github.io/webappsec-mixed-content/#should-block-fetch; see also "We further limit this category..." in https://w3c.github.io/webappsec-mixed-content/#category-optionally-blockable)

This relates to your upgrading blockable content comment below, to be continued there...